packages/dd-trace/src/appsec/rasp/command_injection.js

```diff
@@ -25,19 +25,26 @@ function disable () {
 }
 
 function analyzeCommandInjection ({ file, fileArgs, shell, abortController }) {
-  if (!file || !shell) return
+  if (!file) return
 
   const store = storage.getStore()
   const req = store?.req
   if (!req) return
 
-  const commandParams = fileArgs ? [file, ...fileArgs] : file
-
-  const persistent = {
-    [addresses.SHELL_COMMAND]: commandParams
+  const persistent = {}
+  const raspRule = { type: RULE_TYPES.COMMAND_INJECTION }
+  const params = fileArgs ? [file, ...fileArgs] : file
+
+  if (shell) {
+    persistent[addresses.SHELL_COMMAND] = params
+    raspRule.variant = 'shell'
+  } else {
+    const commandParams = Array.isArray(params) ? params : [params]
+    persistent[addresses.EXEC_COMMAND] = commandParams
+    raspRule.variant = 'exec'
```

Comment:
`addresses.SHELL_COMMAND` address accepts single and array values and `addresses.EXEC_COMMAND` only arrays?

```suggestion
  const params = fileArgs ? [file, ...fileArgs] : [file]

  if (shell) {
    persistent[addresses.SHELL_COMMAND] = params
    raspRule.variant = 'shell'
  } else {
    persistent[addresses.EXEC_COMMAND] = params
    raspRule.variant = 'exec'
```

Reply:
Your proposal is totally right, I pushed a commit for a simple params parsing